gallery_dl/extractor/tiktok.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.tiktok.com/"""
+
+from .common import Extractor, Message
+from .. import exception, text, util, ytdl
+from re import compile, escape, IGNORECASE
+from datetime import datetime
+
+BASE_PATTERN = r"(?:https?://)?(?:www\.)?tiktok(?:v?)\.com"
+USER_PATTERN = BASE_PATTERN + r"/+@([\w.]{0,23}\w)?"
+POST_PATTERN = USER_PATTERN + \
+    r"/+(?:[pP][hH][oO][tT][oO]|[vV][iI][dD][eE][oO])/+(?:[0-9]+)/*"
+VM_POST_PATTERN = r"(?:(?:https?://)?(?:(?:vm|vt)\.)?tiktok\.com/+.*/*)|" + \
+    r"(?:(?:https?://)?(?:www\.)?tiktok\.com/+t/+.*/*)"
+SHARE_PATTERN = BASE_PATTERN + r"/+share/+video/+(?:[0-9]+)/*"
+
+
+class TiktokExtractor(Extractor):
+    """Base class for TikTok extractors"""
+
+    category = "tiktok"
+    directory_fmt = ("{category}", "{user}")
+    filename_fmt = "{title} [{id}{index:?_//}{img_id:?_//}].{extension}"
+    archive_fmt = "{id}_{index}_{img_id}"
+    root = "https://www.tiktok.com/"
+    cookies_domain = ".tiktok.com"
+
+    def urls(self):
+        return [self.url]
```

Comment:
```suggestion
        return [text.ensure_http_scheme(self.url)]
```
`self.url` could be missing a scheme. gallery-dl usually reconstructs URLs instead of (blindly) using user input to avoid this among other things.